Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -103,6 +107,18 @@ cancel bookingId _ req = do
           Right res' -> do
             let merchantOperatingCityId = booking.merchantOperatingCityId
             disToPickup <- driverDistanceToPickup booking.merchantId merchantOperatingCityId (getCoordinates res'.currPoint) (getCoordinates booking.fromLocation)
+            now <- getCurrentTime
+            let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+            distanceRes <-
+              Maps.getDistance booking.merchantId merchantOperatingCityId $
+                Maps.GetDistanceReq
+                  { origin = ride.driverInitialLocation,
+                    destination = getCoordinates res'.currPoint,
+                    travelMode = Just Maps.CAR
+                  }
+            person <- QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+            when ((driverBookingDuration > merchant.driverBookingTime && disToPickup < 50) || (realToFrac distanceRes.distance) > merchant.driverPickupDistance) $ do
```

Comment:
`driverBookingDuration` is in seconds and `driverBookingTime` is stored as minutes in db, can we store in seconds in db too?